### PR TITLE
[MRG+1] Datasets: Uniform representation of labels in MSDL Atlas

### LINKED ISSUE #930: Have a uniform representation of labels for atlases
We should do the work of grabbing the labels of the atlases in the downloaders and expose them uniformely. It would make the life of the users easier.

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Attempts to Fix #930

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
We would like the code to be really interchangeable between Harvard-Oxford and MSDL.

If you look at how Harvard-Oxford is used (for instance in the following example:
http://nilearn.github.io/auto_examples/03_connectivity/plot_signal_extraction.html ) we need the 'labels' field of the data bunch to be an array of labels.

Hence, I suggest putting in this field of the bunch, the "labels" field of the current structure, and a adding two other fields; one "regions_coords" that would be the current "labels[['x', 'y', 'z']].tolist()", and one "network_name" that would contain the "networks" field of the CSV.

As a results, the code should look much more similar for Harvard-Oxford and for MSDL.

**Comment by GaelVaroquaux** [comment]
We also need to make fetch_atlas_aal consistent: the field "regions" should be renamed to "maps", and we should rethink how the names are given. The goal would be to make AAL and Harvard-Oxford completely interchangeable in the examples:

http://nilearn.github.io/modules/generated/nilearn.datasets.fetch_atlas_aal.html#nilearn.datasets.fetch_atlas_aal

**Comment by GaelVaroquaux** [comment]
You have what I believe is a non-spurious failure on numpy 1.6:
https://travis-ci.org/nilearn/nilearn/jobs/110814412

File: examples/03_connectivity/plot_probabilistic_atlas_extraction.py [hunk 1]
```diff
@@ -23,16 +23,9 @@
...
-atlas_filename = atlas['maps']
-
-# Load the labels
-import numpy as np
-csv_filename = atlas['labels']
-
-# The recfromcsv function can load a csv file
-labels = np.recfromcsv(csv_filename)
-names = labels['name']
+atlas_filename, labels = atlas['maps'], atlas['labels']
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_probabilistic_atlas_extraction.py]: I would like these 2 assignments split in 2 different lines: it makes it easier to follow for the beginner.

File: nilearn/datasets/tests/test_atlas.py [hunk 2]
```diff
@@ -229,8 +230,23 @@ def test_fetch_atlas_destrieux_2009():
...
 @with_setup(tst.setup_tmpdata, tst.teardown_tmpdata)
 def test_fetch_atlas_msdl():
+    datadir = os.path.join(tst.tmpdir, 'msdl_atlas')
+    os.mkdir(datadir)
+    os.mkdir(os.path.join(datadir, 'MSDL_rois'))
+    data_dir = os.path.join(datadir, 'MSDL_rois', 'msdl_rois_labels.csv')
+    with open(data_dir, 'w') as csv_file:
+        fieldnames = ['x', 'y', 'z', 'name', 'net_name']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+        writer.writeheader()
```
Comment (GaelVaroquaux) [inline nilearn/datasets/tests/test_atlas.py]: This is the line that's failing in Python 2.6.

Comment (KamalakerDadi) [inline nilearn/datasets/tests/test_atlas.py]: I am not sure whether we really care about Python 2.6 failure here. @AlexandreAbraham any ideas ?

Comment (AlexandreAbraham) [inline nilearn/datasets/tests/test_atlas.py]: We are officially supporting Python 2.6 so the test suite must run in 2.6. It should not be that hard to fix, why not using `np.savetxt` ?

Comment (GaelVaroquaux) [inline nilearn/datasets/tests/test_atlas.py]: > We are officially supporting Python 2.6 so the test suite must run in 2.6. It
> should not be that hard to fix, why not using np.savetxt ?

If np.savetxt does the trick, then let's use it.

However, I am really considering droping Py2.6

**Comment by KamalakerDadi** [comment]
Can I also have reviews on this ?

**Comment by GaelVaroquaux** [comment]
LGTM. This is great. :+1: for merge.

**Pull request merged**